source/extensions/filters/network/zookeeper_proxy/zookeeper_utils.cc

```diff
@@ -0,0 +1,50 @@
+#include "extensions/filters/network/zookeeper_proxy/zookeeper_utils.h"
+
+namespace Envoy {
+namespace Extensions {
+namespace NetworkFilters {
+namespace ZooKeeperProxy {
+
+int32_t BufferHelper::peekInt32(Buffer::Instance& buffer, uint64_t& offset) {
+  int32_t val = buffer.peekBEInt<int32_t>(offset);
+  offset += sizeof(int32_t);
+  return val;
+}
+
+int64_t BufferHelper::peekInt64(Buffer::Instance& buffer, uint64_t& offset) {
+  int64_t val = buffer.peekBEInt<int64_t>(offset);
+  offset += sizeof(int64_t);
+  return val;
+}
+
+bool BufferHelper::peekBool(Buffer::Instance& buffer, uint64_t& offset) {
+  char byte = buffer.peekInt<char, ByteOrder::Host, 1>(offset);
+  bool val = static_cast<bool>(byte & 255);
+  offset += 1;
+  return val;
+}
+
+std::string BufferHelper::peekString(Buffer::Instance& buffer, uint64_t& offset) {
+  std::string val;
+  int32_t len = peekInt32(buffer, offset);
+
+  if (len == 0) {
+    return val;
+  }
+
+  if (buffer.length() < (offset + len)) {
+    throw EnvoyException("buffer is too small");
```

Comment:
probably want to include more context here: eg what are we attempting to parse

there also doesn't seem to be coverage for this line